### PR TITLE
LPS-142856 | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-142856

cc @david-gutierrez-mesa 
cc @Tim-Cao

### CODE REVIEW

**Comment by Tim-Cao** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
Hi @ozysouza, could you resolve conficts?

**Comment by ozysouza** [comment]
> Hi @ozysouza, could you resolve conficts?

Hi @Tim-Cao, conflicts solved. Would you take a look at? Thanks.

**Comment by Tim-Cao** [comment]
ci:test:echo

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, Could you update the usages of NestedPortlets.viewPG? Cases are failed according to [ci:test:echo](https://testray.liferay.com/home/-/testray/case_results/1588156029)

**Comment by ozysouza** [comment]
> Hi @ozysouza, Could you update the usages of NestedPortlets.viewPG? Cases are failed according to [ci:test:echo](https://testray.liferay.com/home/-/testray/case_results/1588156029)

Hi @Tim-Cao, thanks for pointing it out. I updated the cases that were missing it.

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path [hunk 1]
```diff
@@ -95,6 +95,16 @@
 	<td>//div[contains(@class,'modal-content')]//*[@class='modal-title']</td>
 	<td></td>
 </tr>
+<tr>
+	<td>NESTED_APPLICATION_COLUMN</td>
+	<td>//div[contains(@id,'NestedPortletsPortlet')]//div[contains(@class,'portlet-dropzone') and contains(@id,'column-${key_nestedColumnId}')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>NESTED_COLUMN_PORTLET_TITLE</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path:104]: PORTLET_TITLE_INSIDE_NESTED_APPLICATION_COLUMN

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path [hunk 2]
```diff
@@ -150,6 +160,16 @@
 	<td>//div[@id='main-content']//div[contains(@class,'portlet-${key_decorator}')]//h2[contains(@class,'portlet-title') and (.='${key_portletTitleName}')] | //div[@id='main-content']//div[contains(@class,'portlet-${key_decorator}')]//section[contains(@id,'${key_portletTitleName}')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>SPECIFIC_COLUMN_HEADER</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path:164]: HEADER_INSIDE_PAGE_COLUMN

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path [hunk 3]
```diff
@@ -150,6 +160,16 @@
...
+	<td>//div[@id='main-content']//div[contains(@id,'column-${key_headerColumnId}')]//header[contains(@class,'portlet-topper')]//span[.='${key_portletName}']</td>
+	<td></td>
+</tr>
+<tr>
+	<td>SPECIFIC_COLUMN_TITLE</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path:169]: PORTLET_TITLE_INSIDE_PAGE_COLUMN

**Comment by Tim-Cao** [comment]
ci:test:echo

**Comment by Tim-Cao** [comment]
ci:test:frontend

File: portal-web/test/functional/com/liferay/portalweb/macros/Portlet.macro [hunk 4]
```diff
@@ -399,9 +403,20 @@ definition {
 	}
 
 	macro selectApplicationDecorator {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/Portlet.macro:405]: configureApplicationDecorator

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase [hunk 5]
```diff
@@ -20,124 +24,185 @@ definition {
...
+			for (var widgetName : list "Asset Publisher Custom,Documents and Media Custom") {
+				Portlet.viewSpecificPG(portletName = "${widgetName}");
+			}
+		}
 	}
 
 	@description = "This is a test for LPS-98325. Assert that multiple pages with several widget pages(that can be converted) and an api to convert bulk widget pages to content pages is executed, if convertLayouts and getConvertibleLayoutPlid is invoked, an array with the plids of the widget pages that are convertible to content pages is returned and all convertibles widget pages in the site are converted to content pages."
 	@priority = "5"
-	test ConvertMultipleWidgetPagesToContentPagesViaAPI {
+	test MultipleWidgetPagesViaAPI {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase:174]: Keep the original one

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase [hunk 6]
```diff
@@ -227,125 +300,103 @@ definition {
...
+				fragmentName = "heading",
+				text = "Heading Example");
+		}
 	}
 
 	@description = "This is a test for LPS-98325. If there is a site with several widget pages(that can be converted) and some content pages and an api to convert bulk widget pages to content pages is executed, you should be able to convert the widget pages."
 	@priority = "2"
-	test ConvertMultipleWidgetPagesWithContentPagesViaAPI {
-		property test.name.skip.portal.instance = "ContentPageConverter#ConvertMultipleWidgetPagesWithContentPagesViaAPI";
+	test MultipleWidgetPagesWithContentPagesViaAPI {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase:358]: Keep the original one

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase [hunk 7]
```diff
@@ -365,666 +426,856 @@ definition {
...
+		task ("View Asset Publisher is not shown") {
+			PageEditor.viewFragmentNotPresent(fragmentName = "Asset Publisher");
+		}
 	}
 
-	@description = "This is a test for LPS-98325. Assert that widget pages with nested application and customization setting can be converted to content pages via API."
+	@description = "This is a test for LPS-98303 and LPS-98327. Assert convert a widget page with Asset Publisher to content page and the content page keeps the same plid identifier before conversion."
 	@priority = "5"
-	test ConvertWidgetPagesWithNestedApplicationAndCustomizationSettingToContentPagesViaAPI {
+	test WidgetPage {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase:540]: Keep the original one

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase [hunk 8]
```diff
@@ -365,666 +426,856 @@ definition {
...
-				locator1 = "SitePages#LIST_GROUP_ITEM_PAGE_TYPE");
+			AssertElementPresent(locator1 = "PageEditor#FRAGMENT_SIDEBAR_BODY");
 		}
 	}
 
-	@description = "This is a test for LPS-105943. When convert a widget page with one Customizable column, there will be a warning message shown in conversion review"
+	@description = "This is a test for LPS-98589. Multiple widgets with different application decorators survive content page conversion."
 	@priority = "5"
-	test ConvertWidgetPageWithCustomizableColumnContainingWidgets {
+	test WidgetPageDecorators {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase:611]: Keep the original one

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase [hunk 9]
```diff
@@ -365,666 +426,856 @@ definition {
...
+				Portlet.viewApplicationDecorator(
+					decorator = "${decoratorName}",
+					portletName = "${portletName}");
+			}
+		}
+	}
+
+	@description = "This is a test for LPS-98325. Assert that widget pages with nested application and customization setting can be converted to content pages via API."
+	@priority = "5"
+	test WidgetPagesWithNestedApplicationAndCustomizationSettingViaAPI {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase:680]: Keep the original one

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase [hunk 10]
```diff
@@ -365,666 +426,856 @@ definition {
...
+					pageType = "Content Page");
+			}
+		}
 	}
 
-	@description = "This is a test for LPS-105943. When convert a widget page with one Nested Applications widget containing widgets, there will be a warning message shown in conversion preview"
+	@description = "This is a test for LPS-105943. When convert a widget page with one Customizable column, there will be a warning message shown in conversion review"
 	@priority = "5"
-	test ConvertWidgetPageWithNestedApplicationsContainingWidgets {
+	test WidgetPageWithCustomizableColumnContainingWidgets {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase:738]: Keep the original one

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase [hunk 11]
```diff
@@ -365,666 +426,856 @@ definition {
...
-			key_portletName = "Nested Applications",
-			locator1 = "Portlet#TITLE");
+			Portlet.viewNoSpecificPG(portletName = "Asset Publisher");
+		}
+	}
 
-		Button.clickPublish();
+	@description = "This is a test for LPS-105943. When convert a widget page with one Nested Applications widget containing widgets, there will be a warning message shown in conversion preview"
+	@priority = "5"
+	test WidgetPageWithNestedApplicationsContainingWidgets {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase:809]: Keep the orignal one

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase [hunk 12]
```diff
@@ -365,666 +426,856 @@ definition {
...
+			Portlet.viewPG(portletName = "Asset Publisher");
+
+			Portlet.viewNoSpecificPG(portletName = "Nested Applications");
+		}
 	}
 
 	@description = "This is a test for LPS-105943. When convert a widget page with a non-standard layout, there will be a warning message shown in conversion preview"
 	@priority = "5"
-	test ConvertWidgetPageWithNonStandardLayout {
+	test WidgetPageWithNonStandardLayout {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase:901]: Keep the original one

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase [hunk 13]
```diff
@@ -365,666 +426,856 @@ definition {
...
+			}
+		}
 	}
 
-	@description = "This is a test for LPS-105943. When convert a widget page with one Nested Applications widget containing widgets and a non-standard layout, there will be two warning messages shown in conversion preview"
+	@description = "This is a test for LPS-105943. When convert a widget page with one Customizable column and a non-standard layout, there will be two warning messages shown in conversion preview"
 	@priority = "3"
-	test ConvertWidgetPageWithNonStandardLayoutAndNestedApplicationsContainingWidgets {
-		property test.name.skip.portal.instance = "ContentPageConverter#ConvertWidgetPageWithNonStandardLayoutAndNestedApplicationsContainingWidgets";
+	test WidgetPageWithNonStandardLayoutAndCustomizableColumns {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase:1014]: Keep the original one

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase [hunk 14]
```diff
@@ -365,666 +426,856 @@ definition {
...
+			}
 		}
+	}
 
-		Navigator.gotoSitePage(
-			pageName = "Test Page Name",
-			siteName = "Test Site Name");
+	@description = "This is a test for LPS-105943. When convert a widget page with one Nested Applications widget containing widgets and a non-standard layout, there will be two warning messages shown in conversion preview"
+	@priority = "3"
+	test WidgetPageWithNonStandardLayoutAndNestedApplicationsContainingWidgets {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase:1134]: Keep the original one

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase [hunk 15]
```diff
@@ -365,666 +426,856 @@ definition {
...
+			PageEditor.viewNestedElementToGridColumn(
+				column = "1",
+				elementNotPresent = "Nested Applications");
+		}
 	}
 
 	@description = "This is a test for LPS-125933. Convert widget page to content page with a WCD and an Asset Publisher where web contents are displayed."
 	@priority = "3"
-	test ConvertWidgetPageWithWebContentsToContentPage {
+	test WidgetPageWithWebContents {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/ContentPageConverter.testcase:1278]: Keep the original one

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/sitepage/Page.macro [hunk 16]
```diff
@@ -937,25 +937,30 @@ definition {
...
-
-			AssertVisible(locator1 = "Home#PAGE_COLUMN_2_LAYOUT_2_COLUMNS_70_SPLIT");
-		}
-	}
-
 	macro viewPageLayoutColumn {
+		SitePages.gotoPageMenuItem(
+			menuItem = "Configure",
+			pageName = "${pageName}");
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/sitepage/Page.macro:944]: Remove it for reusable and separate the operation and verfication codes

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path [hunk 17]
```diff
@@ -150,6 +160,16 @@
 	<td>//div[@id='main-content']//div[contains(@class,'portlet-${key_decorator}')]//h2[contains(@class,'portlet-title') and (.='${key_portletTitleName}')] | //div[@id='main-content']//div[contains(@class,'portlet-${key_decorator}')]//section[contains(@id,'${key_portletTitleName}')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>SPECIFIC_COLUMN_HEADER</td>
+	<td>//div[@id='main-content']//div[contains(@id,'column-${key_headerColumnId}')]//header[contains(@class,'portlet-topper')]//span[.='${key_portletName}']</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path:165]: Rename key_headerColumnId to key_pageColumnNumber

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path [hunk 18]
```diff
@@ -150,6 +160,16 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>SPECIFIC_COLUMN_TITLE</td>
+	<td>//div[@id='main-content']//div[contains(@id,'column-${key_columnId}')]//h2[contains(@class,'portlet-title') and (.='${key_portletName}')]</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path:170]: Rename key_columnId to key_pageColumnNumber

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path [hunk 19]
```diff
@@ -95,6 +95,16 @@
 	<td>//div[contains(@class,'modal-content')]//*[@class='modal-title']</td>
 	<td></td>
 </tr>
+<tr>
+	<td>NESTED_APPLICATION_COLUMN</td>
+	<td>//div[contains(@id,'NestedPortletsPortlet')]//div[contains(@class,'portlet-dropzone') and contains(@id,'column-${key_nestedColumnId}')]</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path:100]: Rename key_nestedColumnId to key_nestedColumnNumber

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path [hunk 20]
```diff
@@ -95,6 +95,16 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>NESTED_COLUMN_PORTLET_TITLE</td>
+	<td>//section[contains(@id,'NestedPortlets')]//div[contains(@id,'column-${key_nestedColumnId}')]//h2[contains(@class,'portlet-title') and contains(.,'${key_portletName}')]</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/Portlet.path:105]: Rename key_nestedColumnId to key_nestedColumnNumber

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, Could you update according to comments? Thanks

**Comment by Tim-Cao** [comment]
No failure cases are caused by changes in this PR.

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/sitepage/Page.macro (hunk 16 above)
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/macros/layout/sitepage/Page.macro:944]: Ok, I reverted my changes in this macro and effected cases.

**Comment by ozysouza** [comment]
> Hi @ozysouza, Could you update according to comments? Thanks

Hi @Tim-Cao, PR updated. Would you take a look at?

File: portal-web/test/functional/com/liferay/portalweb/paths/util/Home.path [hunk 21]
```diff
@@ -366,43 +366,13 @@
...
-<tr>
-	<td>PAGE_COLUMN_1_LAYOUT_2_COLUMNS_50_SPLIT</td>
-	<td>//div[contains(@class,'portlet-layout')]/div[@id='column-1' and contains(@class,'col-md-6')]</td>
-	<td></td>
-</tr>
-<tr>
-	<td>PAGE_COLUMN_2_LAYOUT_2_COLUMNS_50_SPLIT</td>
-	<td>//div[contains(@class,'portlet-layout')]/div[@id='column-2' and contains(@class,'col-md-6')]</td>
+	<td>PAGE_COLUMN_LAYOUT</td>
+	<td>//div[contains(@id,'main-content')]//div[contains(@class,'col-md-${key_colNumber}') and contains(@id,'column-${key_idNumber}')]</td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/util/Home.path:375]: //div[contains(@id,'main-content')]//div[contains(@class,'col-md-${key_span}') and contains(@id,'column-${key_pageColumnId}')]

**Comment by Tim-Cao** [comment]
Hi @ozysouza, see changes https://github.com/liferay-echo/liferay-portal/compare/fc4713e5d38b834e8f05ab111dab734ee3e57253...547dd54ba861e0395e65fcd6446e7105190eef15

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**